services/github/github-last-commit.service.js

```diff
@@ -57,50 +54,40 @@ module.exports = class GithubLastCommit extends LegacyService {
           repo: 'skia',
           branch: 'infra/config',
         },
-        staticPreview: {
-          label: 'last commit',
-          message: 'april 2018',
-          color: 'yellow',
-        },
+        staticPreview: this.render({ commitDate: '2013-07-31T20:01:41Z' }),
         ...commonExampleAttrs,
       },
     ]
   }
 
-  static registerLegacyRouteHandler({ camp, cache, githubApiProvider }) {
-    camp.route(
-      /^\/github\/last-commit\/([^/]+)\/([^/]+)(?:\/(.+))?\.(svg|png|gif|jpg|json)$/,
-      cache((data, match, sendBadge, request) => {
-        const user = match[1] // eg, mashape
-        const repo = match[2] // eg, apistatus
-        const branch = match[3]
-        const format = match[4]
-        let apiUrl = `/repos/${user}/${repo}/commits`
-        if (branch) {
-          apiUrl += `?sha=${branch}`
-        }
-        const badgeData = getBadgeData('last commit', data)
-        if (badgeData.template === 'social') {
-          badgeData.logo = getLogo('github', data)
-          badgeData.links = [`https://github.com/${user}/${repo}`]
-        }
-        githubApiProvider.request(request, apiUrl, {}, (err, res, buffer) => {
-          if (githubCheckErrorResponse(badgeData, err, res)) {
-            sendBadge(format, badgeData)
-            return
-          }
-          try {
-            const parsedData = JSON.parse(buffer)
-            const commitDate = parsedData[0].commit.author.date
-            badgeData.text[1] = formatDate(commitDate)
-            badgeData.colorscheme = ageColor(Date.parse(commitDate))
-            sendBadge(format, badgeData)
-          } catch (e) {
-            badgeData.text[1] = 'invalid'
-            sendBadge(format, badgeData)
-          }
-        })
-      })
-    )
+  static get defaultBadgeData() {
+    return {
+      label: 'last commit',
+    }
+  }
+
+  static render({ commitDate }) {
+    return {
+      message: formatDate(commitDate),
+      color: ageColor(Date.parse(commitDate)),
+    }
+  }
+
+  async fetch({ user, repo, branch }) {
+    const options = { qs: {} }
+    if (branch) {
+      options.qs.sha = branch
+    }
+    return this._requestJson({
+      url: `/repos/${user}/${repo}/commits`,
```

Comment:
You can write `options: { qs: { sha: branch } }` without a conditional and the result should be the same.